src/Microsoft.AspNetCore.Mvc.Core/Internal/VirtualFileResultExecutor.cs

```diff
@@ -59,14 +69,31 @@ public Task ExecuteAsync(ActionContext context, VirtualFileResult result)
                         count: null,
                         cancellation: default(CancellationToken));
                 }
-                else
+                else if (range == null || !fileStream.CanSeek)
```

Comment:
I spoke to @Eilon and we decided that we will just not support ranges for CanSeek = false. So I treat it like the range was not given and return the entire body